src/algebra/direct_sum/internal.lean

```diff
@@ -101,7 +101,7 @@ direct_sum.to_semiring_of _ _ _ _ _
 
 lemma direct_sum.coe_mul_apply [add_monoid ι] [semiring R] [set_like σ R]
   [add_submonoid_class σ R] (A : ι → σ) [set_like.graded_monoid A]
-  [Π (i : ι) (x : A i), decidable (x ≠ 0)] (r r' : ⨁ i, A i) (i : ι) :
```

Comment:
My objection to this change is that if the user ends up with a weird decidable instance in their goal, they can't use this lemma any more without having to use `convert`.

Reply:
Fortunately it is hard to get a different instance here since `(by classical; apply_instance : decidable (x ≠ 0))` is defeq to `@ne.decidable _ (by classical; apply_instance) x 0`.

We could also change the definitions in dfinsupp to require `decidable (x = 0)` instead of `decidable (x ≠ 0)` to remove the `ne.decidable` diamond.